### DISCUSSION TITLE
Data backup contacts and appointments

### DISCUSSION THREAD

**Discussion by WebworkrNet** [FossifyOrg/General-Discussion#257] [Q&A]
Both the contacts app and the appointment calendar offer data backup.

Only the folder `Intern`> `Download` is available and cannot currently be changed for technical reasons?

I was once told by another developer of another app that file access outside the app directory is not possible at all, if I understood him correctly. How was this technically solved here so that it does work?

When and according to what principle are backups always made? 
I have had both apps installed for months. I have only found a single backup of the calendar. I found several backups of the address book. At first glance, I couldn't recognize any pattern as to when a backup file was created. I frequently make changes to the calendar and especially the address book.

Thank you

**Comment by Aga-C** [comment]
It's already reported: https://github.com/FossifyOrg/General-Discussion/issues/211.

**Comment by WebworkrNet** [comment]
#211 only covers this part of my posting:

> Only the folder Intern> Download is available and cannot currently be changed for technical reasons?

However, the other points are not covered. I therefore ask you to reopen this discussion. Thank you

**Reply by Aga-C** [reply]
> When and according to what principle are backups always made?

Automatic backups are made daily (see [here](https://github.com/FossifyOrg/Contacts/blob/6f3687b8f687557dafd31b5aa9264f7fc2ee3038/app/src/main/kotlin/org/fossify/contacts/helpers/Constants.kt#L17)). If you have one file, you either haven't set placeholders for date in a filename, or it's not working properly on your phone. If you have set a filename like `Events_%Y_%M_%D` and you don't have daily backups, you can report a bug.

> I was once told by another developer of another app that file access outside the app directory is not possible at all, if I understood him correctly. How was this technically solved here so that it does work?

We're using [Storage Access Framework](https://developer.android.com/guide/topics/providers/document-provider).
